bundles/org.openhab.binding.miio/src/test/java/org/openhab/binding/miio/internal/ReadmeHelper.java

```diff
@@ -318,39 +321,40 @@ private MiIoBasicDevice findDatabaseEntry(String deviceName) {
         File dir = new File(path);
-        File[] filesList = dir.listFiles();
+        FileFilter fileFilter = file -> !file.isDirectory() && file.getName().toLowerCase().endsWith(".json");
+        File[] filesList = dir.listFiles(fileFilter);
+        if (filesList == null) {
+            return i18nEntries;
+        }
         for (File file : filesList) {
-            if (file.isFile() && file.getName().toLowerCase().endsWith(".json")) {
-                try {
-                    String key = file.getName().toLowerCase().split("json")[0];
-                    JsonObject deviceMapping = convertFileToJSON(path + file.getName());
-                    Gson gson = new GsonBuilder().serializeNulls().create();
-                    @Nullable
-                    MiIoBasicDevice devdb = gson.fromJson(deviceMapping, MiIoBasicDevice.class);
-                    if (devdb != null) {
-                        for (MiIoBasicChannel channel : devdb.getDevice().getChannels()) {
-                            i18nEntries.put(I18N_CHANNEL_PREFIX + key + channel.getChannel(),
-                                    channel.getFriendlyName());
-                            StateDescriptionDTO state = channel.getStateDescription();
-                            if (state != null) {
-                                List<OptionsValueListDTO> options = state.getOptions();
-                                if (options != null) {
-                                    for (OptionsValueListDTO channelOption : options) {
-                                        String optionValue = channelOption.value;
-                                        String optionLabel = channelOption.label;
-                                        if (optionValue != null && optionLabel != null) {
-                                            i18nEntries.put(
-                                                    I18N_OPTION_PREFIX + key + channel.getChannel() + "-" + optionValue,
-                                                    optionLabel);
-                                        }
+            try {
+                String key = file.getName().toLowerCase().split("json")[0];
+                JsonObject deviceMapping = convertFileToJSON(path + file.getName());
+                Gson gson = new GsonBuilder().serializeNulls().create();
+                @Nullable
+                MiIoBasicDevice devdb = gson.fromJson(deviceMapping, MiIoBasicDevice.class);
+                if (devdb != null) {
+                    for (MiIoBasicChannel channel : devdb.getDevice().getChannels()) {
+                        i18nEntries.put(I18N_CHANNEL_PREFIX + key + channel.getChannel(), channel.getFriendlyName());
+                        StateDescriptionDTO state = channel.getStateDescription();
+                        if (state != null) {
+                            List<OptionsValueListDTO> options = state.getOptions();
+                            if (options != null) {
```

Comment:
Reduce nesting:
```
if (options == null) {
    continue;
}

```

Reply:
to be honest, I did not find this a very nice design pattern, so I made an alternative solution.
Note that this part of the code is not part of the binding itself, but supporting tooling for the binding developer.